frontend/packages/dev-console/src/components/monitoring/alerts/__tests__/MonitoringAlerts.spec.tsx

```diff
@@ -38,11 +40,70 @@ describe('MonitoringAlerts', () => {
+
+  const spyPrometheusRulesPoll = jest.spyOn(prometheusHook, 'usePrometheusRulesPoll');
+  spyPrometheusRulesPoll.mockReturnValueOnce([
+    {
+      data: {
+        groups: [
+          {
+            name: 'kubernetes.rules',
+            rules: [
+              {
+                alerts: [
+                  {
+                    annotations: {
+                      message:
+                        'Alerts are not configured to be sent to a notification system, meaning that you may not be notified in a timely fashion when important failures occur. Check the OpenShift documentation to learn how to configure notifications with Alertmanager.',
+                    },
+                    labels: {
+                      alertname: 'AlertmanagerReceiversNotConfigured',
+                      severity: 'warning',
+                    },
+                    state: 'firing',
+                  },
+                ],
+                annotations: {
+                  message:
+                    'Alerts are not configured to be sent to a notification system, meaning that you may not be notified in a timely fashion when important failures occur. Check the OpenShift documentation to learn how to configure notifications with Alertmanager.',
+                },
+                labels: { prometheus: 'openshift-monitoring/k8s', severity: 'warning' },
+                name: 'AlertmanagerReceiversNotConfigured',
+                query: 'cluster:alertmanager_routing_enabled:max == 0',
+                state: 'firing',
+                type: 'alerting',
+              },
+              {
+                alerts: [],
+                annotations: {
+                  message:
+                    'Cluster Monitoring Operator is experiencing reconciliation error rate of {{ printf "%0.0f" $value }}%.',
+                },
+                labels: { prometheus: 'openshift-monitoring/k8s', severity: 'warning' },
+                name: 'ClusterMonitoringOperatorReconciliationErrors',
+                query:
+                  'rate(cluster_monitoring_operator_reconcile_errors_total[15m]) * 100 / rate(cluster_monitoring_operator_reconcile_attempts_total[15m]) > 10',
+                state: 'inactive',
+                type: 'alerting',
+              },
+            ],
+          },
+        ],
+      },
+    },
+    null,
+    false,
+  ]);
   it('should render monitoring alerts', () => {
     const wrapper = shallow(<MonitoringAlerts {...monitoringAlertsProps} />);
     expect(wrapper.find(FilterToolbar).exists()).toBe(true);
     expect(wrapper.find(Table).exists()).toBe(true);
     expect(wrapper.find(TableHeader).exists()).toBe(true);
     expect(wrapper.find(TableBody).exists()).toBe(true);
   });
+  spyPrometheusRulesPoll.mockReturnValueOnce([{}, null, false]);
+  it('should show empty state message', () => {
+    const wrapper = shallow(<MonitoringAlerts {...monitoringAlertsProps} />);
+    expect(wrapper.find(EmptyBox).exists()).toBe(true);
+  });
```

Comment:
@debsmita1 Really great to see more tests coming. In jest all testcases and code which is required to run a test case should be part of a `it` or `beforeEach`, `afterEach`, etc. function. Only test setup calls which are similar for all tests could be done before all tests.

In your case you modify the `spyPrometheusRulesPoll` mock between two test functions. This works only fine when you run your test from top to bottom 'once'. The first mock call saved one result, the second one the second (empty) version.

The `it` calls just register "test" calls (your callback function) and it will be run later. When you add console.log statements you can see that the tests are runned in this order:

```tsx
console.log('1')
it('test 1' => {
  console.log('3')
})
console.log('2')
it('test 2' => {
  console.log('4')
})
```

And this is an issue if you don't run all tests at the same time or in a different order.  You can test this with this command:

`yarn test packages/dev-console/src/components/monitoring/alerts/__tests__/MonitoringAlerts.spec.tsx -t empty`

This runs both mock "Once" calls and then runs the second it callback. But the mock returns the first value here.

So the general idea should be that each test works completely independent. You can reach this by just moving the setup of `spyPrometheusRulesPoll` also into the it function.